Game/MyProject/Game/RoadGenerator/RoadGeneratorControler.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+#include <algorithm>
+#include <IvEventHandler.h>
+#include <IvVector3.h>
+#include <vector>
+#include <IvGame.h>
+#include "BezierCurve.hpp"
+#include "RoadMargins.hpp"
+#include "../SceneManagement/SceneGraph.hpp"
+#include <iostream>
+
+
+class RoadGeneratorControler
+{
+public:
+    RoadGeneratorControler(std::vector<IvVector3>& bPoints, std::vector<IvVector3>& rMiddle, std::vector<std::pair<IvVector3,IvVector3>>& rMargin):
+    bezierPoints(bPoints), rMiddlePoints(rMiddle), rMarginPoints(rMargin){};
+    ~RoadGeneratorControler(){};
+    
+    void Update(float dt);
+    
+    const std::vector<IvVector3>& getMiddlePoints() const {return rMiddlePoints;};
+    const std::vector<std::pair<IvVector3,IvVector3>>& getMarginPoints() const { return rMarginPoints;};
+    const int getEditIndex() const {return editIndex;};
+    
+private:
+    const float editSpeed = 4.0f;
```

Comment:
Avoid constant members in classes. They just use memory. You can move them outside the class.